spec/integration/basic_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'spec_helper'
+
+describe 'Basic flow' do
+  let(:custom_error_class) { Class.new(StandardError) }
+  let(:custom_error_response) { double(body: custom_response, status: 200) }
+  let(:custom_response) { {custom: 'response'} }
+
+  before do
+    stub_const('CustomErrorClass', custom_error_class)
+    stub_const('CustomErrorResponse', custom_error_response)
+  end
+
+  it 'uses a custom error' do
```

Comment:
Maybe it should be `Describe 'Failure handling' it 'supports customer error mapping'`.